reana_commons/config.py

```diff
@@ -303,6 +303,9 @@ def kubernetes_node_label_to_dict(node_label):
 WORKFLOW_RUNTIME_USER_NAME = os.getenv("WORKFLOW_RUNTIME_USER_NAME", "reana")
 """Default OS user name for running job controller."""
 
+WORKFLOW_RUNTIME_GROUP_NAME = os.getenv("WORKFLOW_RUNTIME_GROUP_NAME", "reana")
```

Comment:
Hmm, I wonder whether we need to create a new group name here at all. We do need to use the GID=0 during runtime for the appropriate workspace file access permissions, but cannot we simply use the default `root` group name directly without introducing a new "reana" group?

```console
$ docker run -i -t --rm docker.io/library/ubuntu:20.04 /bin/bash -c 'grep :0: /etc/group'
root:x:0:
```

That said, it seems nice and consistent to introduce the new variable `WORKFLOW_RUNTIME_GROUP_NAME` instead of using hard-coded "root". But perhaps its default value could be simply "root" and not "reana"? And perhaps we might not actually need to create any such a group and just use the default?

Reply:
Yes, creating a new group is not needed if we keep `WORKFLOW_RUNTIME_USER_GID` set to zero. However that is needed if the GID is not zero, as otherwise the `useradd` call fails.

We can check whether the group already exists or not, and if it doesn't then `groupadd` is called. What do you think?

Reply:
I have updated reana-workflow-controller's PR to only create the group if there is no group with the given GID

Reply:
Sounds good. We may perhaps want to set `WORKFLOW_RUNTIME_GROUP_NAME` to be `root` by default, so that its value would correspond with `WORKFLOW_RUNTIME_USER_GID`'s default value being `0`.

Reply:
Regarding the default name, I would keep `reana` as I see two scenarios:
- gid is zero, so normally the `root` group already exists -> nothing to do
- gid is not zero, so we create a new group `reana`, as we cannot name this `root`
What do you think?

Reply:
After discussing, I have changed `reana` to `root` and I have also added a comment explaining what to do in case the group id needs to be changed.